### PR TITLE
add Ocean Gliders format output option to seaxexplorer processing

### LINKED ISSUE #21: Output OG 1.0
Ultimately, we should aim for the OceanGliders 1.0 format https://github.com/OceanGlidersCommunity/OG-format-user-manual

This will take some time. If you spot an atomic change that can be made to move toward OG 1.0, please submit an Issue for it

### PR DESCRIPTION
**Pull request by callumrollo**
This PR is under development and is waiting on several currently active proposed edits to the Ocean Gliders format before it can be finalised 

This will resolve #21

### CODE REVIEW

**Comment by jklymak** [comment]
A quick skim - I'm not clear that a lot of this cannot be handled via the yaml.  

For the rest, it would be nice to keep new variables like platform id in the yaml as well. I've not looked but perhaps that requires a new variable type that just accepts a constant value.  

Maybe we should have a quick call, but I would rather see a general approach rather than special "og" carve outs

File: pyglider/seaexplorer.py [hunk 1]
```diff
@@ -485,18 +488,26 @@ def raw_to_timeseries(indir, outdir, deploymentyaml, kind='raw',
...
+        outname = f"{outdir}/{ds.attrs['id']}.nc"
+        _log.info('writing %s', outname)
+        ds.to_netcdf(outname, 'w',
+                     encoding={'TIME': {'units': 'seconds since 1970-01-01T00:00:00Z',
+                                        'dtype': 'float64'}})
+    else:
+        outname = outdir + id0 + fnamesuffix + '.nc'
+        _log.info('writing %s', outname)
+        ds.to_netcdf(outname, 'w',
+                     encoding={'time': {'units': 'seconds since 1970-01-01T00:00:00Z',
```
Comment (jklymak) [inline pyglider/seaexplorer.py:509]: For example, 'time'/'TIME' could be replaced by a time-encoder yaml entry. eg calls to `ds.time` could be replaced by `ds[time_var_name]` or similar?  

Not sure about 'latitude' etc, but why do they need to be translated back and forth?  For distance over ground?  Maybe just use inside that function?

File: pyglider/utils.py [hunk 2]
```diff
@@ -633,6 +633,87 @@ def find_gaps(sample_time, timebase, maxgap):
...
+        attrs={
+            "long_name": "model of the glider",
+            "platform_model_vocabulary": "None",
+        },
+    )
+    ds["PLATFORM_SERIAL_NUMBER"] = xr.DataArray(
+        f"sea{ds.attrs['glider_serial'].zfill(3)}",
+        attrs={"long_name": "glider serial number"},
+    )
+    ds["DEPLOYMENT_TIME"] = np.nanmin(ds.TIME.values)
```
Comment (jklymak) [inline pyglider/utils.py:696]: The yamls have a "conversion" field for each variable that tries to fine the method in `utils.py`.  So this _could_ look like 

```
netcdf_variables:
    DEPLOYMENT_TIME:
         "long_name": "date of deployment"
         ...
         "conversion": "nanmin_time"
```

with the accompanying change in `utils.py`.  We may want to think of a way to make that more general `(np.nan_min, "DEPLOYMENT_TIME)` if yaml accepts tuples.

**Comment by VeckoTheGecko** [comment]
66519318 (via #210) introduced a bunch of formatting changes that will likely cause merge conflicts. Here are merge and rebase workflows to avoid these conflicts (it depends on having `pre-commit` installed - see the contributing docs).

I recommend first creating a `temp` branch off of your branch of interest to test these workflows and make sure the output is as expected.

```bash
# merge all the way up to the commit that introduced the formatting changes
git merge 66519318^1

# merge the commit that introduced the formatting changes, prioritising the working branch during conflicts
git merge -s ours 66519318 --no-commit 
pre-commit run --all-files
git commit --no-edit

# merge the rest of the way
git merge main
```

or by doing a rebase (if you want a clean history)

```bash
# Rebase up to the commit introducing formatting changes
git rebase 66519318^1

# Rebase onto the commit introducing formatting changes.
# Accept working branch changes (i.e., "theirs" during rebase
# - different to merge. See git rebase docs) during conflicts.
# During the rebase we will run the formatter after applying
# changes from commits, before committing so that everything is compliant.
git rebase -X theirs -i 66519318
# Replace all "pick" with "edit" or "e". Start the rebase...

# Repeatedly run
git add -u && pre-commit run --all-files && git rebase --continue
# to run pre-commit tooling, stage modified files, and continue
# until rebase is complete. Sometimes pre-commit can't autofix
# issues, then you need to fix them manually.
# If you make a mistake you can always do `git rebase --abort`

# Finally, rebase onto main
git rebase main
```

File: pyglider/seaexplorer.py (hunk 1 above)
Comment (callumrollo) [reply pyglider/seaexplorer.py:509]: I can localise lon/lat to just change within those functions, I'll make a new PR with `ds[time_var_name]` and see what it looks like

**Pull request closed without merging**